File: services/receivables/src/JoaoDeBarro.Receivables.Application/DTOs/ReceivableDto.cs
using JoaoDeBarro.Receivables.Domain;

namespace JoaoDeBarro.Receivables.Application.DTOs;

public class ReceivableDto
{
    public Guid? Id { get; set; }

    public string CustomerName { get; set; } = string.Empty;
    public string ServiceDescription { get; set; } = string.Empty;
    public DateOnly ServiceDate { get; set; }

    public string? ServiceOrderNumber { get; set; }
    public string? InvoiceNumber { get; set; }
    public DateOnly? InvoiceIssueDate { get; set; }

    public DateOnly DueDate { get; set; }
    public string PaymentMethod { get; set; } = string.Empty;

    public decimal GrossAmount { get; set; }
    public decimal IssAmount { get; set; }
    public decimal AmountReceived { get; set; }
    public string CurrencyCode { get; set; } = string.Empty;

    public decimal NetAmount { get; set; }
    public decimal OutstandingAmount { get; set; }
    public string Status { get; set; } = string.Empty;
}

File: WebApps/frontend/src/app/domains/finance/ap-ar/receivables/models/receivable.dto.ts
export interface ReceivableDto{
    id?: string | null;
    customerName: string;
    serviceDescription: string;
    serviceDate: string;
    serviceOrderNumber?: string;
    invoiceNumber?: string;
    invoiceIssueDate?: string;
    dueDate: string;
    paymentDate?: string | null;
    paymentMethod: string;
    grossAmount: number;
    issAmount: number;
    inssAmount: number;
    amountReceived: number;
    currencyCode: string;
    netAmount?: number;
    outstandingAmount?: number;
    status?: string;
}
